Add hipBLASLt examples (#311)
amd/rocm-examples: Libraries/hipBLASLt/README.md

# hipBLASLt Examples

## Summary

> The examples in this subdirectory showcase the functionality of the [hipBLASLt](https://github.com/ROCm/rocm-libraries/tree/develop/projects/hipblaslt) library. The examples build only on Linux for the ROCm (AMD GPU) backend. Note that the examples do not perform any validation; as they're intended to demonstrate how to use the API for different use cases.

Reviewer: ```suggestion
The examples in this subdirectory showcase the functionality of the [hipBLASLt](https://github.com/ROCm/rocm-libraries/tree/develop/projects/hipblaslt) library. The examples build only on Linux for the ROCm (AMD GPU) backend. Note that the examples do not perform any validation. They're intended to demonstrate how to use the API for different use cases.
```

## Prerequisites

### Linux

> - [CMake](https://cmake.org/download/) (at least version 3.21)
> - OR GNU Make - available via the distribution's package manager
> - [ROCm](https://rocm.docs.amd.com/projects/HIP/en/latest/install/install.html) (at least version 7.x.x)

Reviewer: ```suggestion
- [CMake](https://cmake.org/download/) (at least version 3.21).
- Or GNU Make - available via the distribution's package manager.
- [ROCm](https://rocm.docs.amd.com/projects/HIP/en/latest/install/install.html) (at least version 7.x.x).
```

> - [hipBLASLt](https://github.com/ROCm/rocm-libraries/tree/develop/projects/hipblaslt): `hipblaslt` package available from [repo.radeon.com](https://repo.radeon.com/rocm/). The repository is added during the standard ROCm [install procedure](https://rocm.docs.amd.com/projects/HIP/en/latest/install/install.html).

Reviewer: This link goes to HIP installation, not ROCm installation. Is that right?

Author: Yes, the same link is used in all the other examples - might be better to standardise that in a separate PR

### Windows

> Future support for Windows will be included in the future.

Reviewer: ```suggestion
Support for Windows will be included in the future.
```

## Building

### Linux

> Make sure that the dependencies are installed, or use the [provided Dockerfiles](../../Dockerfiles/) to build and run the examples in a containerized environment that has all prerequisites installed.

Reviewer: ```suggestion
Ensure the dependencies are installed or use the [provided Dockerfiles](../../Dockerfiles/) to build and run the examples in a containerized environment that has all prerequisites installed.
```
